cookbooks/vectortile/recipes/default.rb

```diff
@@ -0,0 +1,211 @@
+
+package "postgresql-#{postgresql_version}-postgis-#{postgis_version}"
+
+postgresql_user "tomh" do
+  cluster node[:vectortile][:database][:cluster]
+  superuser true
+end
+
+postgresql_user "pnorman" do
+  cluster node[:vectortile][:database][:cluster]
+  superuser true
+end
+
+postgresql_user "tilekiln" do
+  cluster node[:vectortile][:database][:cluster]
+end
+
+postgresql_user "tileupdate" do
+  cluster node[:vectortile][:database][:cluster]
+end
+
+postgresql_database "tiles" do
+  cluster node[:vectortile][:database][:cluster]
+  owner "tileupdate"
+end
+
+postgresql_schema "tilekiln" do
+  cluster node[:vectortile][:database][:cluster]
+  database "tiles"
+  owner "tileupdate"
+  permissions "tileupdate" => :all, "tilekiln" => :usage
+end
+
+postgresql_database "spirit" do
+  cluster node[:vectortile][:database][:cluster]
+  owner "tileupdate"
+end
+
+postgresql_extension "postgis" do
+  cluster node[:vectortile][:database][:cluster]
+  database "spirit"
+end
+
+# Get a recent osm2pgsql version with backports.
+apt_preference "osm2pgsql" do
+  pin "release o=Debian Backports"
+  pin_priority "600"
+end
+
+apt_package "osm2pgsql"
+
+style_directory = "/srv/vector.openstreetmap.org/spirit"
+git style_directory do
+  repository "https://github.com/pnorman/spirit.git"
+  # Check out head for now
+  # revision "61db723"
+  user "tileupdate"
+  group "tileupdate"
+end
+
+themepark_directory = "/srv/vector.openstreetmap.org/osm2pgsql-themepark"
+git themepark_directory do
+  repository "https://github.com/osm2pgsql-dev/osm2pgsql-themepark.git"
+  user "tileupdate"
+  group "tileupdate"
+end
+
+tilekiln_directory = "/opt/tilekiln"
+
+python_virtualenv tilekiln_directory do
+  interpreter "/usr/bin/python3"
+end
+
+python_package "tilekiln" do
+  python_virtualenv tilekiln_directory
+  python_version "3"
+  version "0.5.1"
+end
+
+template "/srv/vector.openstreetmap.org/html/index.html" do
+  source "index.html.erb"
+  owner "www-data"
+  group "www-data"
+  mode "644"
+end
+
+directory "/srv/vector.openstreetmap.org/data" do
+  user "tileupdate"
+  group "tileupdate"
+  mode "755"
+end
+
+template "/usr/local/bin/import-planet" do
+  source "import-planet.erb"
+  owner "root"
+  group "root"
+  mode "755"
+end
+
+template "/usr/local/bin/tilekiln-storage-init" do
+  source "tilekiln-storage-init.erb"
+  owner "root"
+  group "root"
+  mode "755"
+  variables :tilekiln_bin => "#{tilekiln_directory}/bin/tilekiln", :storage_database => "tiles", :config_path => "#{style_directory}/spirit.yaml"
+end
+
+directory "/srv/vector.openstreetmap.org/complete" do
+  user "root"
+  group "root"
+  mode "755"
+end
+
+execute "tilekiln-storage-init" do
+  command "/usr/local/bin/tilekiln-storage-init"
+  user "tileupdate"
+  not_if { ::File.exist?("/srv/vector.openstreetmap.org/complete/tilekiln-storage-init") }
+  notifies :create, "file[/srv/vector.openstreetmap.org/complete/tilekiln-storage-init]", :immediately
```

Comment:
Does this lock file dance achieve anything useful? Can this not just subscribe to `postgresql_database[tiles]` or `postgresql_schema[schema]` so that is happens when the database is created?